src/history/hs_cursor.c

```diff
@@ -18,56 +18,20 @@ __wt_hs_row_search(WT_CURSOR_BTREE *hs_cbt, WT_ITEM *srch_key, bool insert)
     WT_BTREE *hs_btree;
     WT_DECL_RET;
     WT_SESSION_IMPL *session;
-    bool leaf_found;
-#ifdef HAVE_DIAGNOSTIC
-    WT_PAGE *page;
-#endif
 
     hs_btree = CUR2BT(hs_cbt);
     session = CUR2S(hs_cbt);
-    leaf_found = false;
 
     /*
      * Check whether the search key can be find in the provided leaf page, if exists. Otherwise
      * perform a full search.
      */
-    if (__wt_cursor_page_pinned(hs_cbt, true)) {
-#ifdef HAVE_DIAGNOSTIC
```

Comment:
In datastore, we don't do any search if we have the page pinned. So I removed this part of the code.